assets/src/scripts/store/index.js

```diff
@@ -47,19 +47,28 @@ const getCurrentVariableId = function(timeSeries, variables) {
 
 
 export const Actions = {
+    retrieveLocationTimeZone(latitude, longitude) {
+        return function(dispatch) {
+            return queryWeatherService(latitude, longitude).then(
+                resp => {
+                    const tzIANA = resp.properties.timeZone || 'local'; // set to local timezone if unavailable
+                    dispatch(Actions.setLocationIANATimeZone(tzIANA));
+                }
+            );
+        };
+    },
     retrieveTimeSeries(siteno, params=null) {
         return function (dispatch, getState) {
             const currentState = getState();
             const requestKey = getTsRequestKey('current', 'P7D')(currentState);
             dispatch(Actions.addTimeSeriesLoading([requestKey]));
-
             const timeSeries = getTimeSeries({sites: [siteno], params}).then(
                 series => {
                     const collection = normalize(series, requestKey);
 
                     // Get the start/end times of this request's range.
                     const notes = collection.queryInfo[requestKey].notes;
-                    const endTime = notes.requestDT;
+                    const endTime = new Date(notes.requestDT);
```

Comment:
I wonder if we can figure out startTime without converting to a Date object by subtracting the appropriate number of milliseconds.

Reply:
Or maybe use calcStartTime. We know that we have requested P7D so getting the periodDays from the filter is really not necessary.

Reply:
Opted to use `calcStartTime`.